### PR TITLE
Transformations: Add includeByName to organize fields transformation

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Adding inclusion list filter to organize fields transform.

```
#Existing transform: will hide "Time" field
excludeByName = {
    id: 'organize',
    options: {
        excludeByName: {
            Time: true,
        },
    }
};

#New transform: will hide all fields besides "Time"
includeByName = {
    id: 'organize',
    options: {
        includeByName: {
            Time: true,
        }
    }
};


```

**Why do we need this feature?**
Some contexts we do not have the dataFrame, or there are other situations where we won't want to iterate through all of the fields in the dataframe to specify which ones are excluded. 

**Who is this feature for?**
Grafana developers who have a list of fields they want to show in a dataframe. 

**Special notes for your reviewer:**
This came about as part of my investigation into pulling the new logs table view into the table panel in a dashboard, we're currently saving which columns/fields the user has selected in the explore url state, which can be easily passed into the dashboard panel via transformations. However, those are the fields the user wants to show, and the transform is currently expecting a list of fields we want to exclude.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by oscarkilhed** [comment]
image
When first adding this transformation all the fields looks to be disabled in the UI, but they are all visible in the panel. When one of the fields is enabled, only that field is shown. I think the original functionality of the UI is preferred.

File: public/app/features/transformers/editors/OrganizeFieldsTransformerEditor.tsx [hunk 1]
```diff
@@ -19,10 +19,11 @@ import { useAllFieldNamesFromDataFrames } from '../utils';
 interface OrganizeFieldsTransformerEditorProps extends TransformerUIProps<OrganizeFieldsTransformerOptions> {}
 
 const OrganizeFieldsTransformerEditor = ({ options, input, onChange }: OrganizeFieldsTransformerEditorProps) => {
-  const { indexByName, excludeByName, renameByName } = options;
+  const { indexByName, excludeByName, renameByName, includeByName } = options;
 
   const fieldNames = useAllFieldNamesFromDataFrames(input);
   const orderedFieldNames = useMemo(() => orderFieldNamesByIndex(fieldNames, indexByName), [fieldNames, indexByName]);
+  const filterType = includeByName ? 'include' : 'exclude';
```
Comment (oscarkilhed) [inline public/app/features/transformers/editors/OrganizeFieldsTransformerEditor.tsx:26]: includeByName is defaulting to `{}` which makes the filterType default to include. To preserve the current functionality I think the filterType should default to exclude.

**Comment by oscarkilhed** [comment]
When using this in explore, please add an integration test.

File: public/app/features/transformers/editors/OrganizeFieldsTransformerEditor.tsx (hunk 1 above)
Comment (gtk-grafana) [reply public/app/features/transformers/editors/OrganizeFieldsTransformerEditor.tsx]: Good catch, should be fixed now!

**Comment by gtk-grafana** [comment]
> When using this in explore, please add an integration test.
I've got a line item in the escalation to add an e2e test before GA of the feature that will be using this functionality (logs table): https://github.com/grafana/grafana/issues/77469.
Or are you suggesting that we add an integration test in this PR?

**Comment by oscarkilhed** [comment]
> Or are you suggesting that we add an integration test in this PR?

No, adding it together with the other feature before it goes GA should be fine :)

**Review by oscarkilhed — APPROVED** [review]
:man_dancing:

**Pull request merged**